src/cbc_sdk/credentials.py

```diff
@@ -158,6 +179,66 @@ def to_dict(self):
             _dict[i.name.lower()] = self.get_value(i)
         return _dict
 
+    def get_token_type(self):
+        """
+        Get token type API_KEY or BEARER
+
+        Returns:
+            str: The token type
+        """
+        if self._token_type == "API_KEY":
+            return "API_KEY"
+        elif self._token_type == "API_TOKEN" or self._token_type == "OAUTH_APP":
+            return "BEARER"
+        return "UNKNOWN"
+
+    def get_token(self):
+        """
+        Get token required to authenticate with VMware Carbon Black Cloud
+
+        Returns:
+            str: Token string for VMware Carbon Black Cloud
+        """
+        if self._token_type == "UNKNOWN":
+            return None
+        elif self._token_type == "API_KEY":
+            return self._values[CredentialValue.TOKEN]
+
+        CSP_URL = self._values.get(CredentialValue.CSP_URL_OVERRIDE).rstrip("/")
+
+        if self._token_type == "API_TOKEN":
+            API_TOKEN_URL = f"{CSP_URL}/csp/gateway/am/api/auth/api-tokens/authorize"
+            resp = requests.post(API_TOKEN_URL, {"api_token": self._values[CredentialValue.CSP_API_TOKEN]})
+            json_body = resp.json()
+            if resp.status_code != 200:
+                raise CredentialError(json_body.get("message"))
+
+            # Check that token has scope
+            if json_body.get("scope") is None:
+                raise CredentialError("Access token created with empty scope")
+            return json_body.get("access_token")
+
+        if self._token_type == "OAUTH_APP":
+            OAUTH_APP_TOKEN_URL = f"{CSP_URL}/csp/gateway/am/api/auth/token"
+
+            # Construct Authorization header Basic Base64(client_id:client_secret)
+            client_id = self._values[CredentialValue.CSP_OAUTH_APP_ID]
+            client_secret = self._values[CredentialValue.CSP_OAUTH_APP_SECRET]
+            client_credentials = base64.b64encode(bytes(f"{client_id}:{client_secret}", "utf-8"))
+            headers = {"Authorization": f"Basic {client_credentials.decode('utf-8')}"}
+
+            resp = requests.post(OAUTH_APP_TOKEN_URL, {"grant_type": "client_credentials"}, headers=headers)
+            json_body = resp.json()
```

Comment:
I am not 100% sure, but is it possible that upon certain errors this will raise an exception?

Reply:
I am not aware of non-json errors at the moment but I can add a try except though I am not sure which exception I would want to raise instead

Reply:
I was kind of thinking more like moving this line: `json_body = resp.json()` after the check for != 200

Reply:
I can't because I included the error message in the CredentialError Exception